pkg/cloudprovider/providers/vsphere/vsphere.go

```diff
@@ -305,31 +356,48 @@ func populateVsphereInstanceMap(cfg *VSphereConfig) (map[string]*VSphereInstance
 			glog.Error(msg)
 			return nil, errors.New(msg)
 		}
+
 		for vcServer, vcConfig := range cfg.VirtualCenter {
 			glog.V(4).Infof("Initializing vc server %s", vcServer)
 			if vcServer == "" {
 				glog.Error("vsphere.conf does not have the VirtualCenter IP address specified")
 				return nil, errors.New("vsphere.conf does not have the VirtualCenter IP address specified")
 			}
-			if vcConfig.User == "" {
-				vcConfig.User = cfg.Global.User
-			}
-			if vcConfig.Password == "" {
-				vcConfig.Password = cfg.Global.Password
-			}
-			if vcConfig.User == "" {
-				msg := fmt.Sprintf("vcConfig.User is empty for vc %s!", vcServer)
-				glog.Error(msg)
-				return nil, errors.New(msg)
-			}
-			if vcConfig.Password == "" {
-				msg := fmt.Sprintf("vcConfig.Password is empty for vc %s!", vcServer)
-				glog.Error(msg)
-				return nil, errors.New(msg)
+
+			if !isSecretInfoProvided {
+				if vcConfig.User == "" {
```

Comment:
check for cfg.Global.User is not empty in this block. same for cfg.Global.Password.

Reply:
Done